src/libraries/Microsoft.PowerFx.Interpreter/Environment/UserInfoRecordValue.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx
+{
+    // $$$ Can this be done via TypeMarshaller instead?
+    internal class UserInfoRecordValue : RecordValue
+    {
+        private readonly UserInfo _userInfo;
+
+        public UserInfoRecordValue(UserInfo userInfo, RecordType type)
+            : base(type)
+        {
+            _userInfo = userInfo;
+        }
+
+        public static async Task<FormulaValue> GetUserInfoObject(RecordType userInfoType, IServiceProvider serviceProvider)
+        {
+            var userInfo = serviceProvider.GetService<UserInfo>() ?? throw new InvalidOperationException("UserInfo object was not added to service");
+
+            RecordValue userRecord = new UserInfoRecordValue(userInfo, userInfoType);
+
+            return userRecord;
+        }
+
+        protected override bool TryGetField(FormulaType fieldType, string fieldName, out FormulaValue result)
+        {
+            // Should have called TryGetFieldAsync instead.
+            throw new NotImplementedException();
+        }
+
+        protected override async Task<(bool Result, FormulaValue Value)> TryGetFieldAsync(FormulaType fieldType, string fieldName, CancellationToken cancellationToken)
+        {
+            // $$$ Errors...catch if host object throws
```

Comment:
what is this comment for? what error and catch here?

Reply:
stale, will remove.